### PR TITLE
feature: Remove Parameters associated type from Endpoint protocol

### PR DESCRIPTION
**Pull request by d4nielrve**
Purposed changes in #7

### CODE REVIEW

**Comment by KeatoonMask** [comment]
First of all, thanks @d4nielrve for taking your time for reviewing this new package, what a pleasure! :)

Now, let's discuss about this approach. Modelling the endpoint parameters in the proposed way is something correct to me:
- This solution represents an advantage regarding the current one: it's simpler.
- I see this as a better option in case of implementing a client whose endpoints follow a consistent pattern for parameters.

Nevertheless I do think that current approach provides some additional advantages:
- Better legibility: At the moment of implementing the methods related with the endpoint performing.
- Encapsulation: This promotes separation of concerns, keeping the parameter data contained within the struct.

To illustrate this, let's imagine an endpoint that contains several params to create its path. For example the endpoint that fetches a user pkPass in cadenas app: 
- "https://\(store.host)/\(root)/\(store.idStr)/\(Endpoint.User.qrcode)/\(qrCode)/\(pkpass)"
 
With the proposed approach the method related with the endpoint perform will be:
```
func fetchPkPass(store: ITXStore, root: String, qrCode: String, pkPass: String) async throws -> User {
    try await client.performEndpointRequest(
        endpoint: FetchPkPassEndpoint(store: store, root: root, qrCode: qrCode, pkPass: pkPass)),
        requestBody: .empty
    ).body
}
```  

With the current approach, we can use the Parameter associated type to directly encapsulate all that information, so the method implementation is simpler and cleaner:
```
func fetchPkPass(parameters: FetchPkPassEndpoint.parameters) async throws -> User {
    try await client.performEndpointRequest(
        endpoint: FetchPkPassEndpoint(),
        parameters: parameters,
        requestBody: .empty
    ).body
}
```  

In summary, I see both solutions correct but I see that current approach fits better in real productions apps whose client contains endpoints which require varying and complex parameter structures. 

What do you think?

**Comment by d4nielrve** [comment]
Ok I see, the current approach makes the framework more opinionated, guiding the developer to always create the struct of parameters. In my case I prefer to include in the protocol only those attributes that can be considered as required but obviously it's just an opinion. 
For example in the case you mentioned I actually would create a struct or even nested structs to encapsulate those parameters by domain, but I wouldn't do for those endpoints where you only need 1 or 2 properties whose names are directly related to the endpoint: ```searchUserById -> userId```. 

Let's take a look at a some samples:

In your tests, I need to create the struct for parameters
````
struct MockParameters {
    let name: String
    let age: Int
}
````
and then apply a typealias in the definition of the endpoint because of the associatedtype
````
private struct MockEndpointWithParameters: Endpoint {
    typealias Parameters = MockParameters

    func url(parameters: Parameters) -> URLComponents {
        .init(
            path: mockPath + "/search",
            queryItems: [
                .init(name: "name", value: parameters.name),
                .init(name: "age", value: String(parameters.age))
            ]
        )
    }
}
````
Yes, required attributes are encapsulated into a struct of Parameters, but then you're doing something like
````
let response = try await sut.performEndpointRequest(
            endpoint: MockEndpointWithParameters(),
            parameters: parameters,
            requestBody: .empty
        )
````
In my opinion, the endpoint should include not only those parameters but also the definition of request and response body as you already defined in the protocol, so when you are passing that info in different parameters to the method **performEndpointRequest** it's wrong. The Endpoint should **encapsulate** whatever he needs. So while I'm writting these lines I'm realizing that the method for me should looks something like:
````
let response = try await sut.performEndpointRequest(endpoint: MockEndpointWithParameters(XXX))
````
because we're considering to encapsulate different components.
